sdk/typescript/src/signers/signer-with-provider.ts

```diff
@@ -91,10 +92,16 @@ export abstract class SignerWithProvider implements Signer {
   /**
    * Sign a message using the keypair, with the `PersonalMessage` intent.
    */
-  async signMessage(message: Uint8Array): Promise<SerializedSignature> {
-    return await this.signData(
+  async signMessage(message: Uint8Array): Promise<SignedMessage> {
+    const signature = await this.signData(
       messageWithIntent(IntentScope.PersonalMessage, message),
     );
+
+    return {
+      // TODO: Should this include the intent, or just be the raw bytes that are signed?
```

Comment:
yes we should definitely include the intent bytes, otherwise one might ask you to sign a transaction masqueraded like a personal message.

Reply:
This todo isn't for the message that the user is asked to sign, it's for if the return value includes the intent, or if we just automatically add the intent when verifying the message instead.